Sources/ClientRuntime/Networking/Streaming/ByteStream.swift

```diff
@@ -34,7 +34,7 @@ extension ByteStream: Equatable {
         case (.data(let lhsData), .data(let rhsData)):
             return lhsData == rhsData
         case (.stream(let lhsStream), .stream(let rhsStream)):
-            return lhsStream === rhsStream
+            return try! lhsStream.readToEnd() == rhsStream.readToEnd()
```

Comment:
`try!` is used and the non-async `readToEnd()` method are used, because an implementation of `==()` for Equatable conformance must be non-throwing and non-async.

Since this Equatable conformance is only used in protocol tests, this is okay.

Reply:
I didn't move it, but this extension should be moved out to SmithyTestUtils since it is no longer needed on service clients, only in tests.

Reply:
Apparently it's needed in runtime to do switch case on ByteStream. Will keep it in runtime.

Reply:
E.g., 
```
/Users/runner/work/smithy-swift/aws-sdk-swift/Sources/Core/AWSClientRuntime/Middlewares/Sha256TreeHashMiddleware.swift:56:21: operator function '~=' requires that 'ByteStream' conform to 'Equatable'

              case .empty:
            ~~~ ^
```